src/lib/abstract/abstract.pagination.ts

```diff
@@ -0,0 +1,77 @@
+import { Expose } from 'class-transformer';
+import { IsString, IsOptional } from 'class-validator';
+
+import { PaginationType } from '../interface';
+
+interface PaginationQuery {
+    where: string;
+    nextCursor: string;
+    total: number;
+}
+const encoding = 'base64';
+
+export interface PaginationAbstractResponse<T> {
+    data: T[];
+}
+
+export abstract class AbstractPaginationRequest {
+    private _isNext: boolean = false;
```

Comment:
If I understand correctly, `_isNext` means there are more data to be fetched, then how about naming it `_hasNext`?

Reply:
`_isNext` means whether the request was made with `next page` rather than `get more data`.

Reply:
Got it.
`nextCursor` is used  as `query` [internally](https://github.com/woowabros/nestjs-library-crud/pull/267/files#diff-09cbc5e386f9afd976543e24d4ffab5e983f4d4ca0a9fc8367103aff4f810b93R25)
and `query` is valid JSON then, it means it requests next data,
thus _isNext is set true to store the state to use at [`isNextPage`](https://github.com/woowabros/nestjs-library-crud/pull/267/files#diff-1f8a4f0c3e6b350919ba8b16209de1615718a3366d80993d865ca8f0a2244815R43)

Reply:
In cursor pagination, _isNext means next page,
 but in offset, it mean additionally requested, not next page.

Reply:
@thilllon https://github.com/woowabros/nestjs-library-crud/pull/268